nncf/common/pruning/mask_propagation.py

```diff
@@ -112,7 +112,7 @@ def symbolic_mask_propagation(self, prunable_layers_types: List[str],
 
                     for producer in input_mask.mask_producers:
                         previously_dims_equal = True if can_prune_by_dim[producer] is None \
-                            else can_prune_by_dim[producer]
+                            else can_prune_by_dim[producer].decision
```

Comment:
`PruningAnalysisDesicion` object implements `__bool__` method, so this change is ambiguous